src/test/java/api/loans/OverdueFineScheduledNoticesProcessingTests.java

```diff
@@ -448,9 +466,14 @@ private void assertThatScheduledNoticeExists(TriggeringEvent triggeringEvent, No
   private void assertThatNoticesWereSent(UUID... expectedTemplateIds) {
     List<JsonObject> sentNotices = FakeModNotify.getSentPatronNotices();
 
-    assertThat(sentNotices, hasSize(expectedTemplateIds.length));
+    long numberOfCharges = sentNotices.stream()
+      .map(notice -> notice.getJsonObject("context").getJsonArray("feeCharges"))
+      .map(JsonArray::size)
+      .reduce(0, Integer::sum);
```

Comment:
Add more tests where notices are actually bundled, make sure renewals are not bundled with check-ins, as well as check-ins from different sessions are not mixed